templates/OperationSnippet.stencil

```diff
@@ -10,18 +10,42 @@ public func {{ op.name }}(
     guard let url = self.url(forTemplate: urlTemplate) else { return }
 
     // Construct query
+    {% if op.queryInitParams.count > 0 %}
     var queryParams: [QueryParameter] = [
-        {% for param in op.queryParams %}
-        "{{ param.key }}" : {{ param.value}}
+        {% for param in op.queryInitParams %}
+        ("{{ param.key }}" , {{ param.value}})
         {% endfor %}
     ]
+    {% elif op.queryParams.count > 0 %}
+    var queryParams = [QueryParameter]()
+    {% endif %}
+
+    {% if op.queryParams.count > 0 %}
+    {% for param in op.queryParams %}
+    if {{ param.value }} != nil {
+        queryParams.append("{{ param.key }}", {{ param.value}})
+    }
+    {% endfor %}
+    {% endif %}
 
     // Construct headers
+    {% if op.headerInitParams.count > 0 %}
     var headers = HTTPHeaders([
-        {% for param in op.headerParams %}
-        "{{ param.key }}" : {{ param.value}}
+        {% for param in op.headerInitParams %}
+        {{ param.key }} : "{{ param.value}}",
         {% endfor %}
     ])
+    {% elif op.headerParams.count > 0 %}
+    var headers = HTTPHeaders()
+    {% endif %}
```

Comment:
Same thing here--we want to avoid having if-else logic in our templates, if possible.

Reply:
Removed.